data/transactions/logic/opcodes.go

```diff
@@ -143,53 +150,90 @@ func (d *opDetails) Cost(program []byte, pc int, stack []stackValue) int {
 	return cost
 }
 
-var opDefault = opDetails{linearCost{baseCost: 1}, 1, nil, nil, nil}
-var opBranch = opDetails{linearCost{baseCost: 1}, 3, checkBranch,
-	[]immediate{imm("target", immLabel)}, nil}
+func opDefault() OpDetails {
+	return OpDetails{asmDefault, nil, nil, modeAny, linearCost{baseCost: 1}, 1, nil}
+}
+
+func constants(asm asmFunc, checker checkFunc, name string, kind immKind) OpDetails {
+	return OpDetails{asm, checker, nil, modeAny, linearCost{baseCost: 1}, 0, []immediate{imm(name, kind)}}
```

Comment:
@jannotti Minor and optional:  The other `OpDetails` factory methods rely on `opDefault` for common parameterization.  Is there a reason to deviate here?

* I raise the question because I can imagine a mistake arising when changing `opDefault` and falsely believing it's effectively impacting all usages.
* If there's a by-design reason to avoid `opDefault`, I'd optionally consider leaving an inline comment to minimize chance of unintended change.

If the preference is to keep as is, feel welcomed to resolve the thread.

Reply:
I didn't use opDefaults() because this changed so many fields, it seemed easier to instantiate the object from the basic literal syntax. If opDefaults is ever changed, you'd have to be super careful to see which things modify it anyway, so I don't worry _too_ much, but I concede these constructors are a bit of a hodgepodge.

Reply:
@jannotti Got it - makes sense and resolving.